02-express-tutorial/controllers/people.js

```diff
@@ -0,0 +1,55 @@
+const { people } = require("../data");
+
+const getPeople = (req, res) => {
+    res.json(people);
+};
+
+const addPerson = (req, res) => {
+    try {        
+        const { name } = req.body || {};
+        
+        if (!name) {
+            return res.status(400).json({ success: false, message: "Please provide a name" });
+        }
+
+        const newPerson = { id: people.length + 1, name };
+        people.push(newPerson);
+
+        res.status(201).json({ success: true, data: newPerson });
+    } catch (error) {        
+        console.error("Error in POST /api/v1/people:", error);
+        res.status(500).json({ success: false, message: "An internal server error occurred." });
+    }
+};
+
+const getPersonById = (req, res) => {
+    const person = people.find(p => p.id === parseInt(req.params.id));
```

Comment:
Nesting `parseInt` into this line should still be fine, though normally it is best to validate your inputs before you use them i.e. what will happen if a request sends an invalid ID value?